app/forms/find/v2/subjects/form.rb

```diff
@@ -3,15 +3,16 @@
 module Find
   module V2
     module Subjects
-      class PrimarySubjectsForm
+      class Form
         include ActiveModel::Model
 
-        attr_accessor :subjects
+        attr_accessor :subjects, :context
 
-        validates :subjects, presence: true
+        validates :subjects, presence: { message: ->(form, _data) { I18n.t('.find.v2.subjects.form.blank', type: form.context) } }
 
         def initialize(params = {})
           @subjects = Array(params[:subjects]).compact_blank
+          @context = params[:context]
```

Comment:
I am not sure we should hold context. Can we split into two forms? All the two forms are very simple so I think is fine to split into PrimaryForm and SecondaryForm.